packages/core/src/App/Components/Layout/Header/dtrader-v2/account-switcher-dtrader-v2.tsx

```diff
@@ -0,0 +1,254 @@
+import { getSortedAccountList } from '../../../../Containers/AccountSwitcher/helpers';
+
+type TAccountSwitcherDTraderV2 = RouteComponentProps & {
+    history?: ReturnType<typeof useHistory>;
+};
+
+const AccountSwitcherDTraderV2 = observer(({ history }: TAccountSwitcherDTraderV2) => {
+    const [is_closing, setIsClosing] = React.useState(false);
+
+    const { client, ui, traders_hub } = useStore();
+    const {
+        available_crypto_currencies,
+        accounts,
+        account_list,
+        currency,
+        has_fiat,
+        has_active_real_account,
+        has_any_real_account,
+        has_maltainvest_account,
+        is_eu,
+        is_landing_company_loaded,
+        is_low_risk,
+        is_high_risk,
+        is_logged_in,
+        is_logging_in,
+        is_virtual,
+        loginid: account_loginid,
+        is_mt5_allowed,
+        switchAccount,
+        resetVirtualBalance,
+        upgradeable_landing_companies,
+        real_account_creation_unlock_date,
+    } = client;
+    const { show_eu_related_content, selectRegion, setTogglePlatformType } = traders_hub;
+    const { openRealAccountSignup, setShouldShowCooldownModal, toggleAccountsDialog, toggleSetCurrencyModal } = ui;
+
+    const vrtc_loginid = account_list.find(account => account.is_virtual)?.loginid ?? '';
+    const has_user_set_currency = useHasSetCurrency();
+    const has_cr_account = account_list.find(acc => acc.loginid?.startsWith(BROKER_CODE.CR))?.loginid;
+    const show_separator = is_low_risk && has_maltainvest_account;
+    const show_button =
+        (has_active_real_account && !is_virtual && !is_closing) || (is_virtual && !has_any_real_account);
+
+    const closeAccountsDialog = () => {
+        toggleAccountsDialog(false);
+    };
+
+    const setAccountCurrency = () => {
+        closeAccountsDialog();
+        toggleSetCurrencyModal();
+    };
+
+    const handleSwitchAccount = async (loginid?: string) => {
+        setIsClosing(true);
+
+        if (account_loginid === loginid) {
+            setIsClosing(false);
+            return;
+        }
+
+        closeAccountsDialog();
+        await switchAccount(loginid);
+        setIsClosing(false);
+    };
+
+    const resetBalance = async () => {
+        closeAccountsDialog();
+        await resetVirtualBalance();
+    };
+
+    const canOpenMultiple = () => (available_crypto_currencies.length < 1 && !has_fiat ? true : !is_virtual);
+
+    const getRemainingRealAccounts = (): string[] | [] => {
+        if (show_eu_related_content || is_virtual || !canOpenMultiple() || is_low_risk) {
+            return upgradeable_landing_companies;
+        }
+        return [];
+    };
+
+    const ableToResetBalance = (account: TActiveAccount) => {
+        const account_init_balance = 10000;
+        return !!account?.is_virtual && account?.balance !== account_init_balance;
+    };
+
+    const checkIfUserHaveMoreAccount = (type?: string) =>
```

Comment:
```suggestion
    const hasMoreAccounts = (type?: string) =>
```
since this function returns boolean )